crawl-ref/source/dat/clua/autofight.lua

```diff
@@ -340,7 +340,62 @@ local function get_target(no_move)
   return bestx, besty, best_info
 end
 
+local function target_score(monster_x, monster_y, target_x, target_y)
+  -- TODO: this is a hack using an arbitrary beam spell. Replace with a
+  -- dedicated path function for "normal" projectiles.
+  path = spells.path("Plasma Beam", target_x, target_y, 0,0, true)
```

Comment:
As you've mentioned, we wouldn't want to merge this using Plasma Beam's targeter as a hack. Would suggest a custom beam if there's nothing more appropriate in the quiver code. (There might be?)